src/Tribe/Views/V2/Widgets/Compatibility.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Compatibility for Advanced List Widgets and List Widget.
+ * Scenarios:
+ * * TEC with Pro Activated in V1 ( Reverse )
+ * * Pro Disabled to TEC in V1 ( Default ) - @todo PRO widget removed, free widgets stay
+ * * V1 to V2 ( Default )
+ * * Pro Disabled V2 Active ( Default )
+ * * Pro Active V2 to V1 with constant ( Reverse )
+ * * Pro Disabled V2 back to V1 with constant ( Default ) - @todo PRO widget removed, free widgets stay
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Pro\Views\V2\Widgets
+ */
+
+namespace Tribe\Events\Views\V2\Widgets;
+
+/**
+ * Class Compatibility
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Widgets
+ */
+class Compatibility {
+
+	/**
+	 * The default primary list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $primary_id_base = 'tribe-events-list-widget';
+
+	/**
+	 * The default alternative list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $alternative_id_base = 'tribe-events-adv-list-widget';
+
+	/**
+	 * Adds the filters for V2 Widgets
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'switch_compatibility' ] );
+		add_filter( 'option_sidebars_widgets', [ $this, 'remap_list_widget_id_bases' ] );
+	}
+
+	/**
+	 * Switches the primary and alternative id base when v1 is active
+	 * or v2 widgets are disabled, this enables support for upgrading from
+	 * v1 free list widget to the v1 Pro advanced list widget.
```

Comment:
Please, add a note here about why this method should live in TEC and not PRO, and why.
It's not an immediate intuition and someone with good intentions might think to move it in the future.